src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -522,6 +522,12 @@ private Expression BuildValueResolverFunc(PropertyMap propertyMap, Expression de
                     );
             }
 
+            valueResolverFunc = propertyMap.ValueTransformers
+                .Concat(_typeMap.ValueTransformers)
+                .Concat(_typeMap.Profile.ValueTransformers)
```

Comment:
A few hundred mappings, maybe a few thousand properties, it seems reasonable. For all the existing transformers. This will hurt eventually :)

Reply:
What would we cache? The list could be different since you can define at the property map level.

Alternatively we could push the transformers down from the parent levels, kinda like we do for Include.

Reply:
Yes, but not the transformers per member are the problem, all the rest are. And yes, I thought about smth like your second suggestion, but I'm not sure how that would look like, caching by type seemed more obvious.

Reply:
I think this will only be an issue with large amounts of value transformers. And I'm not sure that would really happen

Reply:
I guess we can wait :)